common/components/org.wso2.carbon.data.agents.log4j/src/main/java/org/wso2/carbon/data/agents/log4j/appender/LogEventAppender.java

```diff
@@ -318,82 +336,28 @@ public void run() {
+            String logTime = logTimeLayout.format(event);
+            String logger = loggerLayout.format(event);
+            String priority = priorityLayout.format(event);
+            String message = messageLayout.format(event);
+            String ip = ipLayout.format(event);
+            String instance = instanceLayout.format(event);
             String stacktrace = "";
-            String ip = "";
-            String instance = "";
-            for (String pattern : patterns) {
-                String currEle = (pattern);
-                TenantAwarePatternLayout patternLayout = new TenantAwarePatternLayout(currEle);
-                if (currEle.equals("%T")) {
-                    tenantID = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%S")) {
-                    serverName = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%A")) {
-                    appName = patternLayout.format(event);
-                    if (appName == null || appName.equals("")) {
-                        appName = "";
-                    }
-                    continue;
-                }
-                if (currEle.equals("%d")) {
 
-                    logTime = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%c")) {
-                    logger = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%p")) {
-                    priority = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%m")) {
-                    message = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%H")) {
-                    ip = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%I")) {
-                    instance = patternLayout.format(event);
-                    continue;
-                }
-                if (currEle.equals("%Stacktrace")) {
-                    if (event.getThrowableInformation() != null) {
-                        stacktrace = getStacktrace(event.getThrowableInformation().getThrowable());
-                    } else {
-                        stacktrace = "";
-                    }
+            if (isStackTrace) {
+                if (event.getThrowableInformation() != null) {
+                    stacktrace = getStacktrace(event.getThrowableInformation().getThrowable());
+                } else {
+                    stacktrace = "";
                 }
             }
             Date date;
             DateFormat formatter;
             formatter = new SimpleDateFormat(LoggingConstants.DATE_TIME_FORMATTER);
             date = formatter.parse(logTime);
```

Comment:
Hi Tishan,

This is fixed in by the commits in https://github.com/wso2/shared-analytics/pull/126

Thanks,
SajithD